### DISCUSSION TITLE
dunsctl `history-pop` and `history-rm` doesn't emit a signal?

### DISCUSSION THREAD

**Discussion by qxb3** [dunst-project/dunst#1277] [Help] [answered]
Hello! I kinda assume that this actions will emit a events in d-bus but i guess not?

```
$ dbus-monitor --profile "interface='org.freedesktop.Notifications'"
#type	timestamp	serial	sender	destination	path	interface	member
#					in_reply_to
sig	1707417506.140839	4294967295	org.freedesktop.DBus	:1.73277	/org/freedesktop/DBus	org.freedesktop.DBus	NameAcquired
sig	1707417506.140853	4294967295	org.freedesktop.DBus	:1.73277	/org/freedesktop/DBus	org.freedesktop.DBus	NameLost
mc	1707417511.756545	6	:1.73318	:1.39	/org/freedesktop/Notifications	org.freedesktop.Notifications	GetServerInformation
mc	1707417511.757148	7	:1.73318	:1.39	/org/freedesktop/Notifications	org.freedesktop.Notifications	GetServerInformation
mc	1707417511.757979	8	:1.73318	:1.39	/org/freedesktop/Notifications	org.freedesktop.Notifications	Notify
```

**Comment by fwsmit** [comment]
Apparently not. What do you need this signal for? You can already request the entire history from dunst

**Reply by qxb3** [reply]
It is useful when i want to get the updated history when i removed a notification. My hacky solution is to just send a invalid signal using `dbus-send`
like this: https://github.com/qxb3/gruvbox.hypr/blob/groove/.config/eww/notification/scripts/notification.sh#L31

**Reply by bynect** [reply]
> It is useful when i want to get the updated history when i removed a notification

why wouldn't simply calling dunstctl history work?

**Reply by qxb3** [reply]
It does work but i want it to get the notification in real time and i want to avoid polling as much as possible

**Reply by bynect** [reply]
I see now that you keep a listen instance always running on the background and use the other subcommands to update that. 
I think that at this point you are better off using normal signals (trap and kill) instead of passing through the dbus route

**Reply by fwsmit** [reply, marked as answer]
I think DBus is well suited for passing signals. For most other things we already pass a signal via DBus, so this just needs to be added.

**Reply by bynect** [reply]
> I think DBus is well suited for passing signals. For most other things we already pass a signal via DBus, so this just needs to be added. 

I'll mark this as a feature

**Comment by bynect** [comment]
In  #1405 I added the signal for history removed and history cleared
